aries_cloudagent/wallet/routes.py

```diff
@@ -582,7 +582,9 @@ async def promote_wallet_public_did(
         multitenant_mgr = profile.inject_or(BaseMultitenantManager)
         # Add multitenant relay mapping so implicit invitations are still routed
         if multitenant_mgr and wallet_id:
-            await multitenant_mgr.add_key(wallet_id, info.verkey, skip_if_exists=True)
+            await multitenant_mgr.get_route_manager(
+                profile, wallet_id
+            ).route_public_did(info.verkey)
```

Comment:
To add to my previous comment. If we bind the route manager to the injection context, we could also remove the multitenant integratino in other parts of the codebase. This would just be:

```py
route_manager = profile.inject(RouteManager)
# Doesn't really do anything in non-multitenancy cases, but means we don't have to explicitly integrate with MT here.
await route_manager.route_public_did(info.verkey)

```